lib/active_model/serializer.rb

```diff
@@ -60,16 +60,41 @@ class << self
 
     # @api private
     def self.serializer_lookup_chain_for(klass)
-      chain = []
+      lookup_chain = []
 
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
+      resource_class_name   = klass.name.demodulize
+      resource_namespace    = klass.name.deconstantize
       serializer_class_name = "#{resource_class_name}Serializer"
 
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
+      # Loops over a namespaced Serializer's name to add multiple
+      # namespaced serializer candidates for lookup in the `lookup_chain` from
+      # the "parent" serializer.
+      #
+      # Example:
+      # if `name.to_s` returns "Api::V1::Admin::UserSerializer" (parent serializer),
+      # and `klass` is Post,
+      # this would make the lookup `lookup_chain`:
+      # [
+      #   "Api::V1::Admin::PostSerializer",
+      #   "Api::V1::PostSerializer",
+      #   "Api::PostSerializer",
+      # ]
+      parent_namespace = name.to_s
 
-      chain
+      if self != ActiveModel::Serializer
+        until parent_namespace.empty?
```

Comment:
I think the way you have it may be the cleanest. 
Just trying to think of a way it could possibly throw an exception.. (I haven't thought of one so far)